src/tox/config/sets.py

```diff
@@ -188,23 +188,17 @@ def register_config(self) -> None:
             desc="the root directory (where the configuration file is found)",
         )
 
-        def work_dir_builder(conf: Config, env_name: str | None) -> Path:  # noqa: U100
-            return (conf.work_dir if conf.work_dir is not None else cast(Path, self["tox_root"])) / ".tox"
-
-        def work_dir_post_process(value: Path) -> Path:
-            return self._conf.work_dir if self._conf.options.work_dir else value
-
         self.add_config(
             keys=["work_dir", "toxworkdir"],
             of_type=Path,
-            default=work_dir_builder,
-            post_process=work_dir_post_process,
+            default=lambda *_: self["tox_root"] / ".tox",  # noqa: U101
```

Comment:
Why are we using here the * operator and not enumerating the argumetns? Why is this now a lambda and not a method as before? I prefered the previous way.

Reply:
The args have no impact on what the function will return. I opted for the star as it allowed a single `noqa` code instead of multiple. 

And a lambda was utilized in the  `temp_dir` config already. I have no strong opinion on lambda/funcs but I think the implementation should be consistent either way.

Will revert!